### PR TITLE
Loop through all features and layers for geom-data match

### PR DESCRIPTION
**Pull request by cKellyDesign**
For onaio/esaro#18

```
LAYERS
~~~~~~
esaro:  layersupdate
gisid:  esaro-18-popup-fix
gisida-react:  filters-phase-2
gisida-react-private:  esaro
```

### CODE REVIEW

File: src/map/addPopUp.js [hunk 1]
```diff
@@ -29,74 +31,84 @@ export default function addMousemoveEvent(mapId, mapboxGLMap, dispatch) {
...
+    for (let f = 0; f < features.length; f += 1) {
+      feature = features[f];
+      layerId = feature && feature.layer && feature.layer.id;
+      layer = layerId && layers[layerId];
+
+      if (layer && layer.type !== 'chart') {
+        data = (layer.aggregate && layer.aggregate.timeseries)
+          && timeseries && timeseries[layerId] && timeseries[layerId].data
+          ? [...timeseries[layerId].data]
+          : layer.source && [...layer.source.data];
```
Comment (KipSigei) [inline src/map/addPopUp.js]: I think we should cater for geojson data here e.g
`(layer.source && (layer.source.data.features || [...layer.source.data]))`

Comment (cKellyDesign) [reply src/map/addPopUp.js]: Very good call!

File: src/map/generateStops.js [hunk 2]
```diff
@@ -148,14 +148,7 @@ export default function (layer, timefield) {
...
   if (layer.aggregate && layer.aggregate.timeseries) {
-    sortedData = rows.sort((a, b) => {
-      if ((a.properties || a)[timefield] > (b.properties || b)[timefield]) {
-        return 1;
-      } else if ((b.properties || b)[timefield] > (a.properties || a)[timefield]) {
-        return -1;
-      }
-      return 0;
-    });
+    sortedData = rows.sort((a, b) => new Date((a.properties || a)[timefield]) - new Date((b.properties || b)[timefield]));
```
Comment (cKellyDesign) [inline src/map/generateStops.js]: @KipSigei Could we check if `new Date(...)` returns a valid date and use the previous sort function if it fails?

Comment (KipSigei) [reply src/map/generateStops.js]: addressed here https://github.com/onaio/gisida/pull/247/commits/755dc1e06e31a6a8af9ba084c0f7deac1cf30f40#diff-3dc2f78ef8f7b4111f5f4a21bae2e552R152

Comment (cKellyDesign) [reply src/map/generateStops.js]: 👍

**Comment by cKellyDesign** [comment]
@KipSigei Can this be merged after #258 is merged?

**Pull request merged**